src/Razor/test/Microsoft.VisualStudio.LanguageServices.Razor.Test/Cohost/CohostOnAutoInsertEndpointTest.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.CodBeAnalysis.Remote.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.AutoInsert;
+using Microsoft.CodeAnalysis.Razor.Settings;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServices.Razor.LanguageClient.Cohost;
+using Microsoft.VisualStudio.Razor.Settings;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+
+public class CohostOnAutoInsertEndpointTest(ITestOutputHelper testOutputHelper) : CohostEndpointTestBase(testOutputHelper)
+{
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_AutoInsertEndTag(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>$0</{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">");
+    }
+
+    [Theory]
+    [InlineData("PageTitle")]
+    [InlineData("div")]
+    [InlineData("text")]
+    public async Task Component_DoNotAutoInsertEndTag_DisabledAutoClosingTags(string startTag)
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <{startTag}>$$
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <{startTag}>
+
+            The end.
+            """;
+
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: ">", autoClosingTags: false, expectResult: false);
+    }
+
+    [Fact]
+    public async Task Component_AutoInsertAttributeQuotes()
+    {
+        var input = $"""
+            This is a Razor document.
+
+            <PageTitle style=$$></PageTitle>
+
+            The end.
+            """;
+        var output = $"""
+            This is a Razor document.
+
+            <PageTitle style="$0"></PageTitle>
+
+            The end.
+            """;
+        await VerifyOnAutoInsertAsync(input, output, triggerCharacter: "=", delegatedResponseText: "\"$0\"");
```

Comment:
nit: Consider inserting a blank line above this method call to be consistent with the tests above. (Same feedback for the tests below.)

Reply:
David suggested just using the string with the parameter name directly in the method call. I tried that and it looks pretty good I think. I'll go with that for now, let me know if you disagree.